ui/src/app/controls/endpoints/endpoints.component.html

```diff
@@ -1,43 +1,49 @@
+            </mat-form-field>
+            <mat-form-field class="quart-width" appearance="outline">
+                <mat-label>Exposure</mat-label>
+                <mat-select [attr.data-cy]="'endpoint-exposure-'+i" formControlName="exposure">
+                    <mat-option value="">(default, public)</mat-option>
+                    <mat-option value="public">public</mat-option>
+                    <mat-option value="internal">internal</mat-option>
+                    <mat-option value="none">none</mat-option>
+                </mat-select>
+            </mat-form-field>
 
-        <mat-form-field class="mid-width" appearance="outline">
-            <mat-label><span>Path</span></mat-label>
-            <input [attr.data-cy]="'endpoint-path-'+i" matInput formControlName="path">
-        </mat-form-field>    
-        <mat-form-field class="quart-width" appearance="outline">
-            <mat-label>Protocol</mat-label>
-            <mat-select [attr.data-cy]="'endpoint-protocol-'+i" formControlName="protocol">
-                <mat-option value="">(default, http)</mat-option>
-                <mat-option value="http">http</mat-option>
-                <mat-option value="https">https</mat-option>
-                <mat-option value="ws">ws</mat-option>
-                <mat-option value="wss">wss</mat-option>
-                <mat-option value="tcp">tcp</mat-option>
-                <mat-option value="udp">udp</mat-option>
-            </mat-select>
-        </mat-form-field>
-        <mat-checkbox [attr.data-cy]="'endpoint-secure-'+i" formControlName="secure">Protocol Is Secure</mat-checkbox>
-    </ng-container>
+            <mat-form-field class="mid-width" appearance="outline">
+                <mat-label><span>Path</span></mat-label>
+                <input [attr.data-cy]="'endpoint-path-'+i" matInput formControlName="path">
+            </mat-form-field>
+            <mat-form-field class="quart-width" appearance="outline">
+                <mat-label>Protocol</mat-label>
+                <mat-select [attr.data-cy]="'endpoint-protocol-'+i" formControlName="protocol">
+                    <mat-option value="">(default, http)</mat-option>
+                    <mat-option value="http">http</mat-option>
+                    <mat-option value="https">https</mat-option>
+                    <mat-option value="ws">ws</mat-option>
+                    <mat-option value="wss">wss</mat-option>
+                    <mat-option value="tcp">tcp</mat-option>
+                    <mat-option value="udp">udp</mat-option>
+                </mat-select>
+            </mat-form-field>
+            <mat-checkbox [attr.data-cy]="'endpoint-secure-'+i" formControlName="secure">Protocol Is Secure</mat-checkbox>
+        </mat-card-content>
+        <mat-card-actions>
+            <button [attr.data-cy]="'endpoint-minus-'+i" mat-icon-button (click)="removeEndpoint(i)">
```

Comment:
You could add a tooltip:

```
matTooltip="delete endpoint"
```

or 

```
[matTooltip]="'delete endpoint '+control.get('name')?.value"
```

Reply:
Done in [`d588446` (#7084)](https://github.com/redhat-developer/odo/pull/7084/commits/d5884465a3cb16b52e9851709f33625d96eeee14)